### ISSUE TITLE
Sound?

### ISSUE THREAD

**Issue by sergix44** [phpgl/visu#9]
Do you have any lib to recommend for sound?
I've tried to compile this for windows but no luck: https://github.com/Ponup/php-openal (seems also a bit outdated)

**Comment by mario-deluna** [comment]
**On the good side:** I'm currently working on a sound implementation directly packaged with php-glfw. You can find the branch here:
https://github.com/mario-deluna/php-glfw/tree/feature/audio

I've built a simple music player with it, which you can find here:
https://github.com/mario-deluna/php-glfw/blob/feature/audio/examples/vg/audio_musicplayer.php

Generally, all you have to do is:

```php
// Initialize the sound engine
$audioEngine = new \GL\Audio\Engine();

$sound = $audioEngine->soundFromDisk('path/to/my/sound.mp3');
$sound->play();
```

It currently supports basic 3D audio, individual volumes, pitching, etc.

**On the bad side:** I have not tested it on Windows yet, but I can provide you with a DLL if you want to try it. :)

**Comment by mario-deluna** [comment]
So the windows build went through, but disclaimer, I have not tested these yet:

[glfw_php8.1_nts_x64.zip](https://github.com/user-attachments/files/17187812/glfw_php8.1_nts_x64.zip)
[glfw_php8.1_ts_x64.zip](https://github.com/user-attachments/files/17187813/glfw_php8.1_ts_x64.zip)
[glfw_php8.2_nts_x64.zip](https://github.com/user-attachments/files/17187814/glfw_php8.2_nts_x64.zip)
[glfw_php8.2_ts_x64.zip](https://github.com/user-attachments/files/17187815/glfw_php8.2_ts_x64.zip)
[glfw_php8.3_nts_x64.zip](https://github.com/user-attachments/files/17187816/glfw_php8.3_nts_x64.zip)
[glfw_php8.3_ts_x64.zip](https://github.com/user-attachments/files/17187817/glfw_php8.3_ts_x64.zip)

**Comment by sergix44** [comment]
> So the windows build went through, but disclaimer, I have not tested these yet:
> 
> [glfw_php8.1_nts_x64.zip](https://github.com/user-attachments/files/17187812/glfw_php8.1_nts_x64.zip) [glfw_php8.1_ts_x64.zip](https://github.com/user-attachments/files/17187813/glfw_php8.1_ts_x64.zip) [glfw_php8.2_nts_x64.zip](https://github.com/user-attachments/files/17187814/glfw_php8.2_nts_x64.zip) [glfw_php8.2_ts_x64.zip](https://github.com/user-attachments/files/17187815/glfw_php8.2_ts_x64.zip) [glfw_php8.3_nts_x64.zip](https://github.com/user-attachments/files/17187816/glfw_php8.3_nts_x64.zip) [glfw_php8.3_ts_x64.zip](https://github.com/user-attachments/files/17187817/glfw_php8.3_ts_x64.zip)

Yup, on windows seems to work!

**Comment by mario-deluna** [comment]
Awesome, please report any issues you encounter. 

This will be released as part of php-glfw v2.3, which I hope won't take much longer, but it obviously still requires documentation and lots of testing :)
